### ISSUE TITLE
`no-shadowed-variable` doesn't flag shadowed variable in for loop

### ISSUE THREAD

**Issue by nchen63** [palantir/tslint#1796]
it should flag the `let i = 34;`

```ts
for (let i = 0; i < 3; i++) {
    let i = 34;
}
```
